sdl_android/src/main/java/com/smartdevicelink/api/screen/SoftButtonManager.java

```diff
@@ -0,0 +1,549 @@
+                if (initialState != null && softButtonObject.getStates() != null) {
+                    for (SoftButtonState softButtonState : softButtonObject.getStates()) {
+                        if (softButtonState != null && softButtonState.getName() != null && softButtonState.getArtwork() != null && !fileManager.hasUploadedFile(softButtonState.getArtwork())) {
+                            if (softButtonState.getName().equals(initialState.getName())) {
+                                initialStatesToBeUploaded.add(softButtonObject.getCurrentState().getArtwork());
+                            } else{
+                                otherStatesToBeUploaded.add(softButtonState.getArtwork());
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+
+        // Upload initial state images
+        if (initialStatesToBeUploaded.size() > 0) {
+            Log.v(TAG, "Uploading soft button initial state artworks");
+            fileManager.uploadArtworks(initialStatesToBeUploaded, new MultipleFileCompletionListener() {
+                @Override
+                public void onComplete(Map<String, String> errors) {
+                    if (errors != null && errors.size() > 0) {
+                        Log.e(TAG, "Error uploading soft button artworks");
+                    }
+                    Log.d(TAG, "Soft button initial artworks uploaded");
+                    update(cachedListener);
+                }
+            });
+        }
+
+
+        // Upload other state images
+        if (otherStatesToBeUploaded.size() > 0) {
+            Log.v(TAG, "Uploading soft button other state artworks");
+            fileManager.uploadArtworks(otherStatesToBeUploaded, new MultipleFileCompletionListener() {
+                @Override
+                public void onComplete(Map<String, String> errors) {
+                    if (errors != null && errors.size() > 0) {
+                        Log.e(TAG, "Error uploading soft button artworks");
+                    }
+                    Log.d(TAG, "Soft button other state artworks uploaded");
+                    // In case our soft button states have changed in the meantime
+                    update(cachedListener);
+                }
+            });
+        }
+
+        // This is necessary because there may be no images needed to be uploaded
+        update(cachedListener);
+    }
+
+    /**
+     * Update the SoftButtonManger by sending a new Show RPC to reflect the changes
+     * @param listener a CompletionListener
+     */
+    protected void update(CompletionListener listener) {
+        cachedListener = listener;
+
+        if (batchUpdates) {
+            return;
+        }
+
+        // Don't send if we're in HMI NONE
+        if (currentHMILevel == null || currentHMILevel == HMILevel.HMI_NONE) {
+            return;
+        }
+
+        Log.v(TAG, "Updating soft buttons");
+
+        cachedListener = null;
```

Comment:
So this actually always gets set to null, what's the point in setting it to begin with in this method?